### ISSUE TITLE
We need more examples code.

### ISSUE THREAD

**Issue by vifird** [llvm/llvm-project#56]
When started to learn LLVM, i found the greatest resistance prevents me from moving forward, is the lacking of example code and code comments. A simple API will cost nearly half a day to found how to use it, and it always be resolved from google and stackoverflow, not from source code, not from official code.
So have you planed to provide more code comments and example code?

**Comment by dcci** [comment]
@vifird Do you have any specific example in mind?

**Comment by vifird** [comment]
Here are my cases:
1. About `getelementptr`  
  I know how write getelementptr by hand, but there seems no example to show how to generate instruction by IRBuilder, it cost me nearly a whole day to find the `IRBuilder::CreateInBoundsGEP` function.
2. About `link c++ function`
  All i found about linking c++ function, is `addGlobalMapping` from stackoverflow, but i cannot use it in version 10, thanks for [this answer](https://github.com/llvm/llvm-project/issues/55), i got `LLJIT::defineAbsolute`.

More questions will appear, i will supplement then.

While, when i learn flutter frame, i can nearly get all i want from source code and code comments, that‘s really convenient and high-authority.

**Comment by gwelymernans** [comment]
This doc should give you a very good understanding of the basic layout and common data structures used by LLVM.

http://llvm.org/docs/ProgrammersManual.html

**Comment by vifird** [comment]
@gwelymernans Had readed it already, helpless.

**Comment by joker-eph** [comment]
In general, I think the [Kaleidoscope tutorial](https://llvm.org/docs/tutorial/MyFirstLanguageFrontend/index.html) is a good starting point to have a good tour of LLVM, but it isn't intended to be exhaustive in terms of visiting all the possible APIs though.

**Comment by vifird** [comment]
Yeah, example code may be the fast way for learning and using the huge number of APIs.
